docs/getting-started/quick-start.md

```diff
@@ -0,0 +1,286 @@
+
+```shell
+openff-bespoke executor retrieve --id "1" --output "acetaminophen-results"
+```
+
+(quick_start_using_api)=
+## Using the API
+
+For the more Python oriented user, or for users who are looking for more control over how the bespoke fit will be
+performed, BespokeFit includes a full Python API.
+
+At the heart of the fitting pipeline is the [`BespokeWorkflowFactory`]. The [`BespokeWorkflowFactory`] encodes the 
+full ensemble of settings that will feed into and control the bespoke fitting pipeline for *any* input molecule, and 
+is used to create the workflows that fully describe how bespoke parameters will be generated *for a specific* molecule:
+
+```python
+from openff.bespokefit.workflows import BespokeWorkflowFactory
+
+factory = BespokeWorkflowFactory()
+```
+
+The default factory will produce [workflows](workflow_chapter) that will augment the ["Parsley"] OpenFF force field 
+with bespoke torsion parameters for all non-terminal *rotatable* bonds in the molecule that have been trained 
+to quantum chemical torsion scan data generated for said molecule using the [Psi4] quantum chemistry 
+package.
+
+:::{note}
+See the [configuration section](quick_start_config_factory) for more info on customising the workflow factory.
+:::
+
+The workflow factory will ingest any molecule that can be represented by the OpenFF Toolkit's [`Molecule`] class
+and produce a [`BespokeOptimizationSchema`] schema:
+
+```python
+//  from openff.bespokefit.workflows import BespokeWorkflowFactory
+//  from openff.toolkit.topology import Molecule
+//  factory = BespokeWorkflowFactory()
+//  
+    input_molecule = Molecule.from_smiles("C(C(=O)O)N") # Glycine
+    
+    workflow_schema = factory.optimization_schema_from_molecule(
+        molecule=input_molecule
+    )
+```
+
+This schema encodes the full workflow that will produce the bespoke parameters for that molecule, including details
+about how any reference QC data should be generated and at what level of theory, the types of bespoke parameters to
+generate and hyperparameters about how they should be trained, and the sequence of fitting steps (e.g. fit a 
+charge model, then re-fit the torsion and valence parameters using the new charge model) that should be performed.
+
+Such a schema is fed into a [`BespokeExecutor`] that will run the full workflow:
+
+```python
+//  from openff.bespokefit.workflows import BespokeWorkflowFactory
+//  from openff.toolkit.topology import Molecule
+//  from openff.qcsubmit.common_structures import QCSpec
+//  factory = BespokeWorkflowFactory()
+//  factory.default_qc_specs = [
+//      QCSpec(
+//          method="gfn2xtb",
+//          basis=None,
+//          program="xtb",
+//          spec_name="xtb",
+//          spec_description="gfn2xtb",
+//      )
+//  ]
+//  input_molecule = Molecule.from_smiles("C(C(=O)O)N") # Glycine  
+//  workflow_schema = factory.optimization_schema_from_molecule(
+//      molecule=input_molecule
+//  )
```

Comment:
What did you think of this style of writing doctests? Do you think we'd be better off just using notebooks, at least for these examples where we're essentially breaking one python session into multiple code examples?

Reply:
Hmm I'd never used this syntax before so it was a little bit foreign to me. My main complaint is that PyCharm does not understand the `//`, and so things like code-completion don't work as well in cells that contain these tests and in general the IDE gets lit up with quite a few errors / warnings that make real errors / warnings harder to spot!

Maybe a notebook would be a better way to go. We can maybe make the switch in a future PR?

Reply:
Maybe we could just use [`testsetup`](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html#directive-testsetup) blocks instead?